src/Constraint/Constraint.php

```diff
@@ -212,4 +215,110 @@ public function __toString()
     {
         return self::$transOpInt[$this->operator] . ' ' . $this->version;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBounds()
+    {
+        $this->extractBounds();
+
+        return $this->bounds;
+    }
+
+    private function extractBounds()
+    {
+        if (null !== $this->bounds) {
+            return;
+        }
+
+        $this->bounds = array(array('lower' => array(), 'upper' => array()));
+
+        // Branches have no bounds
+        if (strpos($this->version, 'dev-') === 0) {
+            return;
+        }
+
+        // Canonicalize version as done internally by PHP for version_compare()
+        $version = preg_replace(
+            array('/[-_+]/', '/([^\d\.])([^\D\.])/', '/([^\D\.])([^\d\.])/'),
+            array('.', '$1.$2', '$1.$2'),
+            $this->version
+        );
+
+        $versionChunks = array_reverse(explode('.', $version));
+
+        if (count($versionChunks) > 5) {
+            throw new \LogicException('Version cannot contain more than 5 parts. Use the VersionParser to normalize the version first.');
+        }
+
+        switch ($this->operator) {
+            case self::OP_EQ:
+                $this->bounds[0]['lower'] = implode('.', array_reverse($versionChunks));
+                $this->bounds[0]['upper'] = implode('.', array_reverse($versionChunks));
+                break;
+            case self::OP_LT:
+                $this->bounds[0]['lower'] = '0';
+                $this->bounds[0]['upper'] = implode('.', array_reverse($this->decreaseVersion($versionChunks)));
+                break;
+            case self::OP_LE:
+                $this->bounds[0]['lower'] = '0';
+                $this->bounds[0]['upper'] = implode('.', array_reverse($versionChunks));
+                break;
+            case self::OP_GT:
+                $this->bounds[0]['lower'] = implode('.', array_reverse($this->increaseVersion($versionChunks)));
+                $this->bounds[0]['upper'] = '9999999.9999999.9999999.9999999.9999999';
+                break;
+            case self::OP_GE:
+                $this->bounds[0]['lower'] = implode('.', array_reverse($versionChunks));
+                $this->bounds[0]['upper'] = '9999999.9999999.9999999.9999999.9999999';
+                break;
+            case self::OP_NE:
```

Comment:
I think you're still thinking about this in a too complex way. We don't need multiple bounds. The lower bound for OP_NE is 0 and the upper bound is 9999999.9999999.9999999.9999999.9999999, that's all. A constraint always has only one lower and one upper bound, there cannot be multiple.

Reply:
Same as above, I want multiple bounds here because that's the correct way to go :) You can still ignore them for your `autoload.php` check case but the library itself should extract all the information so it can also be used for other use cases imho.

Reply:
It is not correct. The whole point of boundaries is to be boundaries. If there are values allowed which are higher than the upper boundary, then it is not an upper boundary.

You're really trying to implement something different from what this ticket is about. It's about finding a lower and an upper boundary, not some kind of interval overlap simplifcation, just boundaries.

If you don't want to implement the ticket, I suggest you make a new one for whatever it is you want to build and make a PR for that, and we just keep the original ticket open, because this is not what this ticket was about.